src/dataflow-types/src/types.rs

```diff
@@ -1145,12 +1145,28 @@ pub mod sources {
     }
 
     #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
-    pub struct KafkaSourceConnector {
+    pub enum ConnectorLiteral {
+        Kafka(KafkaSourceConnectorLiteral),
+    }
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub struct KafkaSourceConnectorLiteral {
         pub addrs: KafkaAddrs,
-        pub topic: String,
         // Represents options specified by user when creating the source, e.g.
         // security settings.
         pub config_options: BTreeMap<String, String>,
+    }
+
+    #[derive(Clone, Debug, Eq, PartialEq, Serialize, Deserialize)]
+    pub enum KafkaSourceConnector {
```

Comment:
Is this maybe a `KafkaConnector` or `KafkaConnection`? The object this signifies is a set of stuff to connect to Kafka, e.g. is this generalizable to sinks?

Reply:
Good call, `KafkaConnector` seems better since it should absolutely be generalizable to sinks pretty trivially.